pkg/controller/machine/controller.go

```diff
@@ -422,21 +422,25 @@ func isInvalidMachineConfigurationError(err error) bool {
 func (r *ReconcileMachine) setPhase(machine *machinev1.Machine, phase string, errorMessage string) error {
 	if stringPointerDeref(machine.Status.Phase) != phase {
 		klog.V(3).Infof("%v: going into phase %q", machine.GetName(), phase)
+		// Patch annotations before updating status.
+		// This will call Patch on the object which will overwrite machine with the current state of the machine
```

Comment:
Do you mean our local copy of the machine?  Is this because we're not already working with a deepCopy?  Should we always be working with a deepCopy?

Reply:
Yes this is the local copy. And we are working with a deepcopy, though, that doesn't make too much sense in this case as we aren't reading from the informer cache directly.

The controller runtime client that we use here always `DeepCopy`s when it get's anything from the cache, so the user doesn't need to. But it also reflects the API response into objects if you call mutating calls (Eg, Patch, Update, Delete) where the API server normally returns the stored object. So by passing `machine` in (our local copy of machine), the client sends the patch request, get's the response from the API server which is the current state of the Machine post the patch, then it unmarshals that response into our `machine` pointer